### PR TITLE
New method, withReadTimeout(), on Interceptor.Chain

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by NightlyNexus** [comment]
but, can you add methods to interfaces without breaking binary compatibility?
though, nobody has a reason to implement Chain (could be an abstract class with controlled constructor in 3.0).

**Comment by dave-r12** [comment]
Good point.

File: okhttp/src/main/java/okhttp3/Interceptor.java [hunk 1]
```diff
@@ -36,5 +37,9 @@
      * of network interceptors; for application interceptors this is always null.
      */
     @Nullable Connection connection();
+
+    int readTimeoutMillis();
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/Interceptor.java]: Are we going to follow this up with the others?

**Comment by JakeWharton** [comment]
Since it's only source-incompatible, I think we can exercise such liberty.

File: okhttp/src/main/java/okhttp3/Interceptor.java (hunk 1 above)
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/Interceptor.java]: Yes

**Comment by swankjesse** [comment]
I’m okay breaking compatibility here. I don’t anticipate user code implementing `Chain`, and I think the net benefit of this change far outweighs the compatibility problem.

I think we follow this up with other options that we can promote from OkHttpClient.

**Pull request merged**